src/System.IO.FileSystem/tests/FileStream/Buffering_regression.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Xunit;
+
+namespace System.IO.FileSystem.Tests
+{
+    public class FileStream_Buffering_regression: FileSystemTest
+    {
+        [Fact]
+        public void FlushSetLengthAtEndOfBuffer()
+        {
+            // This is a regression test for a bug with FileStream’s Flush() 
+            // and SetLength() methods.
+            // The read-buffer was not flushed inside Flush and SetLength when
+            // the buffer pointer was at the end. This causes subsequent Seek 
+            // and Read calls to operate on stale/wrong data.
+
+
+            // customer reported repro
+            using (FileStream fs = new FileStream(GetTestFilePath(), FileMode.Create))
```

Comment:
In general, do we need to clean up after files that are created, or are we ok with leaving the temporary files wherever these are getting created?